examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1405 @@
+    return_value = "TokenMetadataQueryResponse",
+    error = "ContractError"
+)]
+fn contract_token_metadata<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<TokenMetadataQueryResponse> {
+    // Parse the parameter.
+    let params: ContractTokenMetadataQueryParams = ctx.parameter_cursor().get()?;
+    // Build the response.
+    let mut response = Vec::with_capacity(params.queries.len());
+    for token_id in params.queries {
+        // Check the token exists.
+        ensure!(host.state().contains_token(&token_id), ContractError::InvalidTokenId);
+
+        let metadata_url = MetadataUrl {
+            url:  build_token_metadata_url(&token_id),
+            hash: None,
+        };
+        response.push(metadata_url);
+    }
+    let result = TokenMetadataQueryResponse::from(response);
+    Ok(result)
+}
+
+/// Example of implementing a function for receiving transfers.
+/// It is not required to be implemented by the token contract, but is required
+/// to implement such a function by any contract which should receive CIS2
+/// tokens.
+///
+/// This contract function is called when a token is transferred to an instance
+/// of this contract and should only be called by a contract implementing CIS2.
+/// The parameter include a `data` field which can be used to
+/// implement some arbitrary functionality. In this example we choose not to use
+/// it, and define the function to forward any transfers to the owner of the
+/// contract instance.
+///
+/// Note: The name of this function is not part the CIS2, and a contract can
+/// have multiple functions for receiving tokens.
+///
+/// It rejects if:
+/// - Sender is not a contract.
+/// - It fails to parse the parameter.
+/// - Contract name part of the parameter is invalid.
+/// - Calling back `transfer` to sender contract rejects.
+#[receive(contract = "cis2_multi_royalties", name = "onReceivingCIS2", error = "ContractError")]
+fn contract_on_cis2_received<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    // Ensure the sender is a contract.
+    let sender = if let Address::Contract(contract) = ctx.sender() {
+        contract
+    } else {
+        bail!(CustomContractError::ContractOnly.into())
+    };
+
+    // Parse the parameter.
+    let params: OnReceivingCis2Params<ContractTokenId, ContractTokenAmount> =
+        ctx.parameter_cursor().get()?;
+
+    // Build the transfer from this contract to the contract owner.
+    let transfer = Transfer {
+        token_id: params.token_id,
+        amount:   params.amount,
+        from:     Address::Contract(ctx.self_address()),
+        to:       Receiver::from_account(ctx.owner()),
+        data:     AdditionalData::empty(),
+    };
+
+    let parameter = TransferParams::from(vec![transfer]);
+
+    // Send back a transfer
+    host.invoke_contract_read_only(
```

Comment:
Can you add a test to the unit tests below that invokes the `onReceivingCIS2` hook?

Reply:
Isn't this function only called when the receiver is a contract - the receiver is assumed to be an account in this algorithm so the logic for this function has not been changed from the cis2-multi sample?